proto/greenfield/storage/events.proto

```diff
@@ -553,3 +554,12 @@ message EventCompleteMigrationBucket {
   // The src_primary_sp_id defines the primary sp id of the bucket before migrate.
   uint32 src_primary_sp_id = 5;
 }
+
+message EventSetTag {
+  // resource_type define the type of the source that the tag belongs to
+  greenfield.resource.ResourceType resource_type = 1;
+  // resource_id define the u256 id of the source that the tag belongs to
+  string resource_id = 2;
+  // tags define the tag of the source
+  map<string, string> tags = 3;
```

Comment:
Can you check how proto buffer encodes `map`? 

We need to ensure it can produce stable outputs, this is the base for safety usage.

Reply:
Changed